### PR TITLE
add content for deep path traversals

### PR DESCRIPTION
**Pull request by matea16**
add content for deep path traversals

### CODE REVIEW

File: pages/querying/best-practices.mdx [hunk 1]
```diff
@@ -26,7 +26,232 @@ Indexing best practices in general + index hinting + analyze graph as strategies
...
+`total_USD` node property, traversing only across `CLOSE_TO` relationships and 
+return the result as a graph, use the following query:
+
+```cypher
+MATCH path=(n {id: 0})-[:CLOSE_TO *WSHORTEST (r, n | n.total_USD)]-(m {id: 15})
+RETURN path;
+```
+
+**When to use weighted shortest?** <br/>
+Use a weighted shortest path algorithm when you need to find the **shortest path**
```
Comment (katarinasupe) [inline pages/querying/best-practices.mdx:98]:
```suggestion
Use the weighted shortest path algorithm when you need to find the **shortest path** 
```

File: pages/querying/best-practices.mdx [hunk 2]
```diff
@@ -26,7 +26,232 @@ Indexing best practices in general + index hinting + analyze graph as strategies
...
+To find the weighted shortest path between nodes based on the value of the 
+`total_USD` node property, traversing only across `CLOSE_TO` relationships and 
+return the result as a graph, use the following query:
+
+```cypher
+MATCH path=(n {id: 0})-[:CLOSE_TO *WSHORTEST (r, n | n.total_USD)]-(m {id: 15})
+RETURN path;
+```
+
+**When to use weighted shortest?** <br/>
```
Comment (katarinasupe) [inline pages/querying/best-practices.mdx:97]:
```suggestion
**When to use WSP?** <br/>
```

File: pages/querying/best-practices.mdx [hunk 3]
```diff
@@ -26,7 +26,232 @@ Indexing best practices in general + index hinting + analyze graph as strategies
...
+similar-weighted paths, and this algorithm fetches them all.
+
+The following query searches for all shortest paths with a default weight equal to 1:
+
+```cypher
+MATCH path=(n {id: 0})-[:CloseTo *ALLSHORTEST (r, n | 1)]-(m {id: 15})
+RETURN path;
+```
+
+**When to use all shortest?** <br/>
```
Comment (katarinasupe) [inline pages/querying/best-practices.mdx:122]:
```suggestion
**When to use ASP?** <br/>
```

File: pages/querying/best-practices.mdx [hunk 4]
```diff
@@ -26,7 +26,232 @@ Indexing best practices in general + index hinting + analyze graph as strategies
...
+Use an all shortest paths algorithm when you need to find all shortest paths between
```
Comment (katarinasupe) [inline pages/querying/best-practices.mdx:123]:
```suggestion
Use the all shortest paths algorithm when you need to find all shortest paths between 
```

File: pages/querying/best-practices.mdx [hunk 5]
```diff
@@ -26,7 +26,232 @@ Indexing best practices in general + index hinting + analyze graph as strategies
...
+### Optimizing graph traversals
+Deep path traversal algorithms play a crucial role in exploring and analyzing 
+graph data. In Memgraph, a powerful graph database management system, optimizing 
+these algorithms can significantly enhance performance when dealing with 
+large-scale graph structures. 
+
+Let's dive into some strategies for optimizing traversal algorithms in Memgraph, 
+along with providing examples using Cypher queries.
+
+For example, let's take a following unoptimized query that traverses through the
```
Comment (katarinasupe) [inline pages/querying/best-practices.mdx:147]:
```suggestion
For example, let's take the following unoptimized query that traverses through the 
```

File: pages/querying/best-practices.mdx [hunk 6]
```diff
@@ -26,7 +26,232 @@ Indexing best practices in general + index hinting + analyze graph as strategies
...
+
+```cypher
+MATCH path=(n)-[*BFS]->(m)
+WHERE n:City AND n.name = 'London' AND m:City 
+RETURN path;
+```
+
+By not using the inline filtering, the database first scans all nodes and then additionally
+applies filter using the `WHERE` clause, causing longer scanning time and slower 
+performance.
```
Comment (katarinasupe) [inline pages/querying/best-practices.mdx:174]: Nice!

File: pages/querying/best-practices.mdx [hunk 7]
```diff
@@ -26,7 +26,232 @@ Indexing best practices in general + index hinting + analyze graph as strategies
...
+
+This query creates an index on the property of nodes, enabling faster lookups 
+during algorithm traversals.
+
+<Cards>
+    <Card
+    title="Read more about indexing"
+    href="/fundamentals/indexes"
+    />
+</Cards>
```
Comment (katarinasupe) [inline pages/querying/best-practices.mdx:238]: I would put this at the beginning of optimizations. Wdyt?

File: pages/querying/best-practices.mdx [hunk 8]
```diff
@@ -26,7 +26,232 @@ Indexing best practices in general + index hinting + analyze graph as strategies
...
+The following query will only return the results if the path is equal to or shorter
+than 2 hops:
+
+```cypher
+MATCH path=(:City {name: 'London'})-[r:ROAD *BFS ..2 (r, n | r.continent = 'Europe')]->(:City)
+RETURN path;
+```
+
+By knowing the schema of your dataset and filtering and limiting the wanted results, 
+you can achieve much more optimized way of using the traversal algorithms
```
Comment (katarinasupe) [inline pages/querying/best-practices.mdx:254]:
```suggestion
you can achieve much more optimized way of using the traversal algorithms.
```

File: pages/querying/best-practices.mdx (hunk 6 above)
Comment (antejavor) [reply pages/querying/best-practices.mdx:174]: The same query, different query plan if there is an index inside, the query planner will look for all filters (`WHERE`) and replace them with index calls, so it is smart in a sense.

**Review by antejavor — CHANGES_REQUESTED** [review]
Small note.

**Pull request merged**